hummingbot/connector/exchange/bloxroute_openbook/bloxroute_openbook_order_data_manager.py

```diff
@@ -112,27 +44,24 @@ async def ready(self):
 
     @property
     def is_ready(self):
-        return self._is_ready
-
-    @property
-    def started(self):
-        return self._started
+        return self._ready.is_set()
 
     async def start(self):
-        if not self._started:
-            self._started = True
+        await self._start_lock.acquire()
+
+        if not self._started.is_set():
+            self._started.set()
```

Comment:
BTW I think you don't actually need the `start_lock` for this. Python coroutines run asynchronously, but they only ever release execution at `await` keywords and actual execution is single threaded. So you don't need to worry about parallelism like you see in Go, where another function might try to execute a critical section if you don't have it locked.

So basically, I'd suggest using only one of `_start_lock` and `_started`, but not both

Reply:
I see, that's good to know - so you are saying that because `_started` is set before the await function is called, if a second coroutine were to call `start`, it would already be set?

Reply:
yeah, pretty much.

```python
start = False
v = 1+1
print(start) # always False

start = False
await v()
print(start) # could be True
```